app/ante/evm/eth.go

```diff
@@ -191,6 +201,20 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 			return ctx, errorsmod.Wrapf(err, "failed to verify the fees")
 		}
 
+		// Check if the fees exceed the spendable account balance.
+		// TODO: Is this only giving the spendable balance or also the locked balance?
+		from := msgEthTx.GetFrom()
```

Comment:
Those code seems duplicated (from the decorator)

Reply:
not sure what you mean? these lines exist only once in the AnteHandle function in my repo

Reply:
`deductFeesFromBalanceOrUnclaimedRewards` has the same logic

Reply:
ahh yes it's slightly different once in the EVM AnteHandler and once in the Cosmos AnteHandler, but could be better refactored